### ISSUE TITLE
clarify: nacl port, or nacl inspired?

### ISSUE THREAD

**Issue by dominictarr** [ttaubert/nacl-browser#1]
This isn't actually compatible with nacl, since it does not use the same primitives.
Although, it does follow the api patterns of nacl, and that is a pretty good thing, I feel it would be more clear to say it's _nacl inspired_ than it's a _port_ because that implies it's compatible with nacl.

**Comment by ttaubert** [comment]
Yes, it is indeed not compatible as the WebCrypto API sadly doesn't (yet?) provide all those fancy algorithms. The original nacl pages do refer to a few fallback algorithms, mostly marked as "TODO" which WebCrypto supports so I'm using those. [1] I didn't touch this code for months and it badly needs to be updated and/or clarified. The "Public-key cryptography" section of nacl is missing too.

I am still not sure whether the original nacl API "for network communication, encryption, ..." is the right fit for a JS module. The need to securely generate nonces and IVs, and store them separately seems like a source of possible failures. Although there surely might be settings where developers want to implement a protocol where both sides keep track of nonces and they never need to be stored or transmitted. Maybe the library should just like PyNaCl provide a random() utility function and provide the right nonce and key sizes via constants. 

[1] https://github.com/ttaubert/nacl-browser/blob/master/README.md#notes
